.changelog/20251212_1132.txt

```diff
@@ -0,0 +1,4 @@
+type:fix
+Isolate role policy restrictions in query planner
+This regression was introduced in my previous fix (https://github.com/cerbos/cerbos/pull/2831). After this "fix", role policy restrictions were aggregated and applied globally to the query plan. This caused a regression where restrictions on a specific role incorrectly constrained permissions granted by other unrestricted roles held by the user.
```

Comment:
I think this should be simplified a bit. It exists purely for the benefit of end users who are reading the release notes. So, something along the lines of "fixes a regression introduced as a result of fixing how role policy effects are intersected for principals with multiple roles" should suffice.